eng/pipelines/templates/stages/source-build-and-validate.yml

```diff
@@ -89,7 +89,8 @@ stages:
           pool: ${{ parameters.pool_Linux }}
         buildSourceOnly: true
         ${{ if eq(parameters.useMicrosoftBuildAssetsForTests, false) }}:
-          extraProperties: "/p:DisableDevBuildAsDefaultForSourceOnly=true"
+          ${{ if contains(variables['Build.DefinitionName'], '-source-build') }}:
+            extraProperties: "/p:DisableDevBuildAsDefaultForSourceOnly=true"
```

Comment:
Can you add a comment here as to why this is necessary?

Reply:
https://github.com/dotnet/source-build/issues/5109#issuecomment-2867158302 suggests it is a temporary workaround. I thought this PR is actually is supposed to remove it or would it be removed after this change is consumed?

Reply:
That kind of what I was wondering. Let's see what @mthalman has to say.

Reply:
It was meant to support the standalone source build pipeline. But it's no longer necessary now that https://github.com/dotnet/dotnet/pull/515 is merged.